site/documentation/content/user-guide/coap-adapter.md

```diff
@@ -0,0 +1,238 @@
+
+## CoAP Content Formats
+
+CoAP doesn't use a textual identifier for content types. Instead numbers are used, which are maintained by the [IANA](https://www.iana.org/).
+For the currenlty assigned numbers, please check [IANA - CoAP Content Formats](https://www.iana.org/assignments/core-parameters/core-parameters.xhtml#content-formats).
+The use of unsupported content types in messages sent to the device are ignored!
+
+## Publish Telemetry Data (authenticated Device)
+
+The device is authenticated using PSK.
+
+* URI: `/telemetry`
+* Method: `POST`
+* Type:
+  * `CON`: *at least once* (`1`) QoS levels
+  * `NON`: *at most once* (`0`) QoS levels
+* Request Options:
+  * (optional) `content-format`: The type of payload contained in the request body. Required, if request contains payload.
+  * (optional) `URI-query: hono-ttd`: The number of seconds the device will wait for the response.
+  * (optional) `URI-query: empty`: If the request doesn't contain payload.
+* Request Body:
+  * (optional) Arbitrary payload encoded according to the given content type. Maybe empty, if `URI-query: empty` is provided.
+* Response Options:
+  * (optional) `content-format`: A media type describing the semantics and format of payload contained in the response body. This option will only be present if the response contains a command to be executed by the device which requires input data.
+  * (optional) `location-query: hono-command`: The name of the command to execute. This option will only be present if the response contains a command to be executed by the device.
+  * (optional) `location-path: ('command'|'command_response/'<command-request-id>)`: The location path contains `'command'` for one-way-commands, or `'command_response' / <command-request-id>`  for commands expecting  a response. In case of a expected response, the location-path contains exactly the URI-path expected for the response.
+* Response Body:
+  * (optional) Arbitrary data serving as input to a command to be executed by the device, if status code is 2.05 (CONTENT).
+  * (optional) Error details, if status code is >= 4.00.
+* Status Codes:
+  * 2.04 (CHANGED): The telemetry data has been accepted for processing. Note that if the message type is `NON` (*at most once* semantics),
+    this status code does **not** mean that the message has been delivered to any potential consumer. However, if the message type is `CON` (*at least once* semantics), then the adapter waits for the message to be delivered and accepted by a downstream consumer before responding with this status code.
+  * 2.05 (CONTENT): The telemetry data has been accepted for processing. The response contains a command for the device to execute.
+  * 4.00 (Bad Request): The request cannot be processed. Possible reasons for this include:
+        * The content format option is missing.
```

Comment:
... but the request body is not empty. However, the code suggests that in this case a 4.06 (Not Acceptable) is being returned ...

Reply:
Done